geoh5py/shared/entity.py

```diff
@@ -212,6 +214,53 @@ def create(cls, workspace, **kwargs):
         )
         return new_object
 
+    @property
+    def coordinate_reference_system(self) -> dict | None:
+        """
+        Coordinate reference system attached to the entity.
+        """
+        coordinate_reference_system = None
+
+        if self._metadata is not None:
+            coordinate_reference_system = self._metadata.get(
+                "Coordinate Reference System", None
+            )
+
+            if coordinate_reference_system is not None:
+                coordinate_reference_system = coordinate_reference_system.get(
+                    "Current", None
+                )
+
+        if coordinate_reference_system is None:
+            coordinate_reference_system = DEFAULT_CRS
```

Comment:
```suggestion
        if self._metadata is not None and "Coordinate Reference System" in self._metadata:
           coordinate_reference_system = self._metadata["Coordinate Reference System"]["Current"]
```

Reply:
it will raise an error if "Current" not in the coordinate system. 
Accepted and changed to:

        coordinate_reference_system = DEFAULT_CRS

        if self._metadata is not None and "Coordinate Reference System" in self._metadata:
            coordinate_reference_system = self._metadata["Coordinate Reference System"].get(
                "Current", DEFAULT_CRS)

Reply:
Good thinking. Not sure if it's ever allowed to have coordinate system without "Current", but that's safer for sure.